modules/imgproc/src/color_hsv.simd.hpp

```diff
@@ -269,17 +283,17 @@ struct RGB2HSV_f
         v_float32 v_max_rgb = v_max(v_max(v_r, v_g), v_b);
 
         v_float32 v_eps = vx_setall_f32(FLT_EPSILON);
-        v_float32 v_diff = v_max_rgb - v_min_rgb;
-        v_s = v_diff / (v_abs(v_max_rgb) + v_eps);
-
-        v_float32 v_r_eq_max = v_r == v_max_rgb;
-        v_float32 v_g_eq_max = v_g == v_max_rgb;
-        v_h = v_select(v_r_eq_max, v_g - v_b,
-              v_select(v_g_eq_max, v_b - v_r, v_r - v_g));
-        v_float32 v_res = v_select(v_r_eq_max, (v_g < v_b) & vx_setall_f32(360.0f),
+        v_float32 v_diff = v_sub(v_max_rgb, v_min_rgb);
+        v_s = v_div(v_diff, v_add(v_abs(v_max_rgb), v_eps));
+
+        v_float32 v_r_eq_max = v_eq(v_r, v_max_rgb);
+        v_float32 v_g_eq_max = v_eq(v_g, v_max_rgb);
+        v_h = v_select(v_r_eq_max, v_sub(v_g, v_b),
+              v_select(v_g_eq_max, v_sub(v_b, v_r), v_sub(v_r, v_g)));
+        v_float32 v_res = v_select(v_r_eq_max, v_and(v_lt(v_g, v_b), vx_setall_f32(360.0f)),
```

Comment:
> `v_and(v_lt(v_g, v_b), vx_setall_f32(360.0f)`

v_select() should be used instead of bits manipulation.

BTW, comparison functions and `v_select()` should work through dedicated masks type (`vbool32_t` in case of RVV) instead of the same data type.
This is old legacy design bug - comparison returns the same type as inputs type.
I believe it should be redesigned during adding of new "scalable" API.